core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -2452,11 +2454,17 @@ public static String getDataTypeBasedOnPrecision(int precision, int scale)  {
     }
   }
 
-  private static int adjustPrecisionIfNeeded(String dataType, int precision) {
-    if (dataType.equals("BIGNUMERIC") && precision > 38) {
-      return 38;
+  private static int[] adjustPrecisionAndScaleIfNeeded(String dataType, int precision, int scale) {
+    int[] precisionScale = {precision, scale};
+    if (dataType.equals("BIGNUMERIC")) {
+      if (precision > 38) {
```

Comment:
check if we can rewrite this method as below , also after modifying this check for case when precision and scale having value 38.
private static int[] adjustPrecisionAndScaleIfNeeded(String dataType, int precision, int scale) {
    final int MAX_PRECISION_SCALE = 38;
    if ("BIGNUMERIC".equals(dataType)) {
        precision = Math.min(precision, MAX_PRECISION_SCALE);
        scale = Math.min(scale, MAX_PRECISION_SCALE);
    }
    return new int[] {precision, scale};
}

Reply:
Done